### DISCUSSION TITLE
Please keep useFieldArray id at least as a feature

### DISCUSSION THREAD

**Discussion by wilav-dev** [react-hook-form/react-hook-form#9247] [General]
Hello, I'm combining useFieldArray with dnd-kit to allow draggable items. Each item must have a unique id in order to be a draggable item (index is not working). If you remove internal id, I must iterate over all my items to add an id. Maybe you can leave it as a feature?

Thank you.

**Comment by bluebill1049** [comment]
What do you mean keep id? i don't think we are removing them with useFieldArray.

**Reply by wilav-dev** [reply]
keyName: Name of the attribute with autogenerated identifier to use as the key prop. This prop is no longer required and will be removed in the next major version.

**Reply by bluebill1049** [reply]
oh, the prop will be removed not the actual id.

**Reply by yamila-fraiman** [reply]
And how will be called the attribute? id? In my system when I have an ID attribute I treat it as a reference so I really need to be able to rename for example using __id

**Reply by bluebill1049** [reply]
it will only be inside the field object, `field.id` and it's not going to be part of your form values. if you need to store your own id, you can include that inside `defaultValues` or inside `append` action, or simply use a hidden input.
